plugin/src/main/groovy/com/btkelly/gnag/tasks/GnagReportTask.java

```diff
@@ -133,8 +139,18 @@ private String getPRSha() {
 
   private void updatePRStatus(GitHubStatusType gitHubStatusType) {
     getLogger().debug("Updating PR Status to: " + gitHubStatusType);
-    if (StringUtils.isNotBlank(getPRSha())) {
-      gitHubApi.postUpdatedGitHubStatus(gitHubStatusType, getProject().getName(), getPRSha());
+    if (useGitHubStatuses) {
+      if (StringUtils.isNotBlank(getPRSha())) {
+        gitHubApi.postUpdatedGitHubStatus(gitHubStatusType, getProject().getName(), getPRSha());
+      }
+    } else {
+      if (gitHubStatusType == ERROR || gitHubStatusType == FAILURE) {
+        if (failOnError) {
+          throw new GradleException(gitHubStatusType.getDescription());
+        } else {
+          throw new StopExecutionException(gitHubStatusType.getDescription());
```

Comment:
This is cool! I didn't even know Gradle had a special exception to stop the task but not actually fail it. 🥇